nixos/modules/hardware/corectrl.nix

```diff
@@ -0,0 +1,40 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.corectrl;
+in {
+
+  options.programs.corectrl = {
+    enable = mkEnableOption ''
+      A tool to overclock amd graphics cards and processors.
+      Add your user to the corectrl group to run corectrl without it asking for your password
+    '';
+
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.corectrl ];
+
+    services.dbus.packages = [ pkgs.corectrl ];
+
+    security.polkit.extraConfig = ''
+polkit.addRule(function(action, subject) {
+    if ((action.id == "org.corectrl.helper.init" ||
+         action.id == "org.corectrl.helperkiller.init") &&
+        subject.local == true &&
+        subject.active == true &&
+        subject.isInGroup("corectrl")) {
+            return polkit.Result.YES;
+    }
+});
+    '';
+
+    boot.kernelParams = [ "amdgpu.ppfeaturemask=0xffffffff" ];
```

Comment:
Please explain this parameter and the magical constant (in the form of a comment).